barretenberg/cpp/src/barretenberg/plonk_honk_shared/execution_trace/execution_trace_usage_tracker.hpp

```diff
@@ -84,25 +84,23 @@ struct ExecutionTraceUsageTracker {
         for (auto [max_size, fixed_block] : zip_view(max_sizes.get(), fixed_sizes.get())) {
             size_t start_idx = fixed_block.trace_offset;
             size_t end_idx = start_idx + max_size;
-            active_ranges.push_back(Range{ start_idx, end_idx });
+            active_ranges.emplace_back(start_idx, end_idx);
```

Comment:
I think this will not break the mac build since Range is just a std::pair which probably has an explicit constructor but you might want to make sure that's right. What you have is obviously better form but I'm suspicious as to why it was using the push_back syntax previously.. may have just been because the Range struct used to be something custom.

Reply:
To be clear the mac build will break if you try to use emplace_back(<constructor_inputs>) for a struct which may be trivially constructible but does not have an explicitly defined constructor. For example it would not work if you had 
```
struct Range {
    size_t start;
    size_t end;
};

Reply:
ohh interesting, i will double check this is ok